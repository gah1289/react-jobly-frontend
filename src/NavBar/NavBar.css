.Nav a {
	color: #fff;
	text-decoration: none;
	padding: 0 2vw;
}

.Nav a:hover {
	color: bisque;
	text-decoration: underline;
}

.hidden {
	display: none;
}
