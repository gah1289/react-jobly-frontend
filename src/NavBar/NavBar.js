import React, { useContext, useState, useEffect } from 'react';
import './NavBar.css';
import { NavLink } from 'react-router-dom';
import { Navbar, Nav, NavItem, NavbarBrand } from 'reactstrap';
import ItemContext from '../ItemContext';
import JoblyApi from '../api';

function NavBar() {
	// The navbar list the Jobly logo as a link for the home page. It will also list NavLinks for companies,
	// const { token, isLoggedIn } = useContext(ItemContext);

	const { isLoggedIn } = useContext(ItemContext);

	return (
		<div>
			<Navbar className="my-2 " color="dark" dark>
				<NavbarBrand href="/">Jobly</NavbarBrand>
				<Nav className="Nav">
					<NavItem>
						<NavLink to="/companies">Companies</NavLink>
					</NavItem>
					<NavItem>
						<NavLink to="/jobs">Jobs</NavLink>
					</NavItem>

					{isLoggedIn && (
						<NavItem>
							<NavLink to="/profile">Profile</NavLink>
						</NavItem>
					)}
					{isLoggedIn && (
						<NavItem>
							<NavLink to="/logout">Log Out</NavLink>
						</NavItem>
					)}
				</Nav>
			</Navbar>
		</div>
	);
}

export default NavBar;
