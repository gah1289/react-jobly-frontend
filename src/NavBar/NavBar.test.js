import React from 'react';
import { render, screen } from '@testing-library/react';
import NavBar from './NavBar';
import { MemoryRouter } from 'react-router';
import ItemContext from '../ItemContext';

describe('Renders Routes', function() {
	let isLoggedIn = true;

	let isLoading = false;

	let user = {
		username     : 'testUser',
		firstName    : 'Test',
		lastName     : 'User',
		email        : 'test@user.com',
		password     : 'password',
		isAdmin      : false,
		applications : []
	};

	user.current = user;
	it('renders without crashing', function() {
		render(
			<MemoryRouter>
				<ItemContext.Provider value={{ isLoading, isLoggedIn, user }}>
					<NavBar />
				</ItemContext.Provider>
			</MemoryRouter>
		);
	});

	it('matches snapshot', function() {
		const { asFragment } = render(
			<MemoryRouter>
				<ItemContext.Provider value={{ isLoading, isLoggedIn, user }}>
					<NavBar />
				</ItemContext.Provider>
			</MemoryRouter>
		);
		expect(asFragment()).toMatchSnapshot();
	});

	it('shows links if user logged in', function() {
		render(
			<MemoryRouter>
				<ItemContext.Provider value={{ isLoading, isLoggedIn, user }}>
					<NavBar />
				</ItemContext.Provider>
			</MemoryRouter>
		);
		const profileLink = screen.queryByText(/Profile/i);
		expect(profileLink).toBeInTheDocument();
		const logoutLink = screen.queryByText(/Log out/i);
		expect(logoutLink).toBeInTheDocument();
	});
	it('Only shows Company and Job links if user logged out', function() {
		user.current = null;
		isLoggedIn = false;
		render(
			<MemoryRouter>
				<ItemContext.Provider value={{ isLoading, isLoggedIn, user }}>
					<NavBar />
				</ItemContext.Provider>
			</MemoryRouter>
		);
		const companiesLink = screen.queryByText(/Companies/i);
		expect(companiesLink).toBeInTheDocument();
		const jobsLink = screen.queryByText(/Jobs/i);
		expect(jobsLink).toBeInTheDocument();
		const profileLink = screen.queryByText(/Profile/i);
		expect(profileLink).not.toBeInTheDocument();
		const logoutLink = screen.queryByText(/Log out/i);
		expect(logoutLink).not.toBeInTheDocument();
	});
});
