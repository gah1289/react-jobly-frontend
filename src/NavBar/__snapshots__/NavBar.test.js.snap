// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Routes matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="my-2  navbar navbar-dark bg-dark"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Jobly
        </a>
        <ul
          class="Nav nav"
        >
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/companies"
            >
              Companies
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/jobs"
            >
              Jobs
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/profile"
            >
              Profile
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/logout"
            >
              Log Out
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
